### PR TITLE
[gpinitsystem2.0] Functional tests for expansion support

### PR DESCRIPTION
**Pull request by srividyaky**
This commit adds functional tests for expansion support changes pushed in the  PR 
https://github.com/greenplum-db/gpdb/pull/17335

Below changes are included:

init_cluster_config_validation_test.go - This contains tests related to the input expansion config validations, new functions GetDefaultExpansionConfig is added to create default config with expansion support along with config validaiton related tests

init_cluster_db_validation_test.go - This contains tests related to the database configuration checks after the cluster is successfully initialized with expansion support, it also included validation for spread and group mirroring

init_cluster_test.go: This includes expansion validation by  passing config file with  different file formats

init_cluster_suite_test.go: In this file check has been added to fetch the hostname

### CODE REVIEW

File: gpMgmt/bin/go-tools/test/integration/init_cluster/init_cluster_config_validation_test.go [hunk 1]
```diff
@@ -398,6 +398,249 @@ func TestInputFileValidation(t *testing.T) {
 			t.Errorf("got %q, want %q", result.OutputMsg, expectedOut)
 		}
 	})
+	t.Run("validate expansion with no coordinator details", func(t *testing.T) {
```
Comment (c0dehu1k) [inline gpMgmt/bin/go-tools/test/integration/init_cluster/init_cluster_config_validation_test.go:401]: configFile := testutils.GetTempFile(t, "config.json")
		config := GetDefaultExpansionConfig(t)

		if err := config.WriteConfigAs(configFile); err != nil {
			t.Fatalf("failed to write config to file: %v", err)
		}

		result, err := testutils.RunInitCluster(configFile)
		if e, ok := err.(*exec.ExitError); !ok || e.ExitCode() != 1 {
			t.Fatalf("got %v, want exit status 1", err)
		}

These lines of code are repeated in every test case.. We should move it to utility functions.

**Review by c0dehu1k — CHANGES_REQUESTED** [review]

File: gpMgmt/bin/go-tools/test/integration/init_cluster/init_cluster_config_validation_test.go [hunk 2]
```diff
@@ -398,6 +398,249 @@ func TestInputFileValidation(t *testing.T) {
...
+			t.Fatalf("got %v, want exit status 1", err)
+		}
+
+		expectedOut := "[ERROR]:-invalid mirroring-Type: test_mirror. Valid options are 'group' and 'spread'"
+		if !strings.Contains(result.OutputMsg, expectedOut) {
+			t.Errorf("got %q, want %q", result.OutputMsg, expectedOut)
+		}
+	})
+
+	t.Run("verify expansion without mirror support", func(t *testing.T) {
```
Comment (piyushc01) [inline gpMgmt/bin/go-tools/test/integration/init_cluster/init_cluster_config_validation_test.go:616]: Expansion when mirrors are also specified.

Comment (piyushc01) [inline gpMgmt/bin/go-tools/test/integration/init_cluster/init_cluster_config_validation_test.go:616]: Expansion when mirror-base-port specified and not specified. 
Returns error when an expansion parameter is present and segment-array is also specified.

Comment (piyushc01) [reply gpMgmt/bin/go-tools/test/integration/init_cluster/init_cluster_config_validation_test.go:616]: detects multi-home when provided

Comment (piyushc01) [reply gpMgmt/bin/go-tools/test/integration/init_cluster/init_cluster_config_validation_test.go:616]: returns error when multi-home env, host have different number of interfaces e.g.:
sdw1: sdw1-1, sdw1-2
sdw2: sdw2-1, sdw2-2, sdw2-3
sdw3: sdw3-1, sdw3-2, sdw3-3

**Comment by srividyaky** [comment]
updated in new PR

https://github.com/greenplum-db/gpdb/pull/17431

**Pull request closed without merging**